### PR TITLE
Allow configuring port of http probes

### LINKED ISSUE #30547: Allow configuring the port for the http probe
### Description

Currently, there is no configuration option for setting the port of the `httpAction` for probes. The port number is infered from the build system, by looking up for a port named `http`.

There are cases where we need to use another port (e.g. the management port). We should provide the option in `PortConfig` to use a differnet port name or value.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #30547

### CODE REVIEW

**Comment by iocanel** [comment]
cc @cescoffier

**Comment by geoand** [comment]
I'll defer to @Sgitario who has been doing a lot of work in this area

**Review by cescoffier — APPROVED** [review]
LGTM, I should be able to change the port when the management interface is used by producing my own decorator.

**Comment by iocanel** [comment]
> LGTM, I should be able to change the port when the management interface is used by producing my own decorator.

This is something you can do already, without this pull request. The pull request is more about exposing control to the user via configuration properties. 

From your comment I feel that you prefer to control it from an extension. This can be done by registering your own decorator, but probe port handling differs in same platforms (e.g. knative) and we wouldn't want to expose that to the management bits. So, it should be cleaner if we had a different built it, that would just allow as select the probe port name. Let me add it.

**Comment by cescoffier** [comment]
In my context, at build time, I know if the management interface is enabled or not. If yes, then the health checks (and others) are not exposed on the main HTTP server (using the `http` port) but use a separate server exposed on the `management` port.

My idea was to update the extension exposing routes to the management interface and passing the port name as you did for `http`.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
Comment (cescoffier) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:686]: That's awesome! I just have to emit that build item if the management interface is used.

Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:686]: That's exactly the idea.

File: extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java
Comment (Sgitario) [inline extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java:6]: I think this sentence is incomplete

**Review by Sgitario — COMMENTED** [review]
I'm ok with the changes, but I have a suggestion that I think it's cleaner. Let me know what you think about.

File: extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java
Comment (iocanel) [reply extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java]: What is the suggestion?

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java
Comment (cescoffier) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java:19]:
```suggestion
        //The problem is that the ApplyDeploymentTrigger decorator depends on a lot of irrelevant Decorators
```

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java
Comment (cescoffier) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java:20]:
```suggestion
        //So, this one needs to inherit the same decorator or we can't guarantee that its executed after
```

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java
Comment (cescoffier) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java:14]:
```suggestion
     * The port number to use when configuring the {@literal http get} action.
```

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java
Comment (cescoffier) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java:21]:
```suggestion
     * The port name for selecting the port of the {@literal HTTP get} action.
```

File: extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java
Comment (cescoffier) [inline extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesProbePortNameBuildItem.java:6]:
```suggestion
 * A build item for selecting which port to use for probes using an {@literal HTTP get} action.
```

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java
Comment (cescoffier) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java:15]:
```suggestion
     * If not configured, the port corresponding to the {@code httpActionPortName} will be used.
```

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Comment by cescoffier** [comment]
@iocanel I guess, the failures are related. WDYT?

**Comment by iocanel** [comment]
> @iocanel I guess, the failures are related. WDYT?

They must be. I'll have a look later today

**Review by Sgitario — CHANGES_REQUESTED** [review]
I think we should address the issue in ChangeDeploymentTriggerDecorator and RemoveDeploymentTriggerDecorator in a separated pull request since it's nothing to do with allowing configuring ports of HTTP probes. 
Let's address this unrelated issue in https://github.com/quarkusio/quarkus/pull/30566

**Comment by iocanel** [comment]
> I think we should address the issue in ChangeDeploymentTriggerDecorator and RemoveDeploymentTriggerDecorator in a separated pull request since it's nothing to do with allowing configuring ports of HTTP probes. Let's address this unrelated issue in #30566

I don't really mind which is the PR that deals with it. It can be #30768 too. One thing that is not clear to me is how are you able to reproduce the issue in #30768? I've only been able to reproduce the issue in this PR.

**Comment by Sgitario** [comment]
> > I think we should address the issue in ChangeDeploymentTriggerDecorator and RemoveDeploymentTriggerDecorator in a separated pull request since it's nothing to do with allowing configuring ports of HTTP probes. Let's address this unrelated issue in #30566
> 
> I don't really mind which is the PR that deals with it. It can be #30768 too. One thing that is not clear to me is how are you able to reproduce the issue in #30768? I've only been able to reproduce the issue in this PR.

That's a good question. It seems not possible to always reproduce this issue using Quarkus CI, but it is in:
- Dekorate Main: https://github.com/dekorateio/dekorate/actions/runs/4071499845
- Using the jakarta-rewrite branch from Quarkus
- and I'm using a basic Quarkus example with OpenShift + Docker

**Comment by cescoffier** [comment]
@iocanel Any ETA on this one?

**Comment by iocanel** [comment]
> @iocanel Any ETA on this one?

We'll have a dekorate release that will unblock this one later today!

**Comment by iocanel** [comment]
@cescoffier: just bumped the PR. Hopefully it will pass now.

**Review by cescoffier — APPROVED** [review]

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java
Comment (Sgitario) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java:29]: Do we need this change?

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java
Comment (Sgitario) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java:15]: Do we need this change?

**Review by Sgitario — CHANGES_REQUESTED** [review]
Changes look good to me. 
Only two questions remain. I think the changes in the deploymenttrigger decorators are not needed.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java
Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java:29]: Yes we do. Pretty much all remove decorators are meant to be processes after the apply ones

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java
Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java:15]: Same goes for change decorators. 
No, things may work even without those changes but it will be circumstantial.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java
Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java:29]: I am wrong, we don't need this change, however I think that it's ok to have.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java
Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ChangeDeploymentTriggerDecorator.java:15]: Same. I am wrong, let me completely wipe them.

**Comment by cescoffier** [comment]
The CI issues look related.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java
Comment (Sgitario) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RemoveDeploymentTriggerDecorator.java:29]: I agree that semantically it's the same, but I would avoid making changes that are unrelated to the issue itself (which is about configuring the port of HTTP probes).

**Comment by cescoffier** [comment]
@iocanel ping?

**Comment by iocanel** [comment]
> @iocanel ping?

Will update the PR now

**Comment by gsmet** [comment]
Please, if you don't run CI locally, create a draft PR and run CI in your fork first.

**Review by Sgitario — APPROVED** [review]

**Review by Sgitario — CHANGES_REQUESTED** [review]
Sorry, I approved it too soon without realising that https://github.com/quarkusio/quarkus/pull/30566#pullrequestreview-1293212226 was not addressed yet.

**Comment by iocanel** [comment]
> Sorry, I approved it too soon without realising that [#30566 (review)](https://github.com/quarkusio/quarkus/pull/30566#pullrequestreview-1293212226) was not addressed yet.

You didn't approve the PR too soon. I did drop the commit completely, but I think that I accidentially pulled it back.

**Comment by Sgitario** [comment]
> > Sorry, I approved it too soon without realising that [#30566 (review)](https://github.com/quarkusio/quarkus/pull/30566#pullrequestreview-1293212226) was not addressed yet.
> 
> You didn't approve the PR too soon. I did drop the commit completely, but I think that I accidentially pulled it back.

The change in [RemoveDeploymentTriggerDecorator.java](https://github.com/quarkusio/quarkus/pull/30566/files#diff-ab2a71e5e7d44be41fd2b87e43b6d6afbab3186b906023f94d601cd86498ee4a) still remains.

**Comment by iocanel** [comment]
> > > Sorry, I approved it too soon without realising that [#30566 (review)](https://github.com/quarkusio/quarkus/pull/30566#pullrequestreview-1293212226) was not addressed yet.
> > 
> > 
> > You didn't approve the PR too soon. I did drop the commit completely, but I think that I accidentially pulled it back.
> 
> The change in [RemoveDeploymentTriggerDecorator.java](https://github.com/quarkusio/quarkus/pull/30566/files#diff-ab2a71e5e7d44be41fd2b87e43b6d6afbab3186b906023f94d601cd86498ee4a) still remains.

It should be ok now.

**Review by Sgitario — APPROVED** [review]
lgtm!

**Pull request merged**
